packages/libs/components/src/stories/plots/NetworkPlot.stories.tsx

```diff
@@ -171,6 +177,120 @@ WithActions.args = {
   getNodeMenuActions: getNodeActions,
 };
 
+// with node label control
+const SimpleWithControlData = genNetwork(
+  20,
+  true,
+  true,
+  DEFAULT_PLOT_SIZE,
+  DEFAULT_PLOT_SIZE
+);
+
+export const SimpleWithControl: Story<TemplateProps> = () => {
+  // list of node labels
+  const labels = SimpleWithControlData.nodes
+    .flatMap((node) => {
+      return node.label != null ? node.label : undefined;
+    })
+    .filter((data) => data != null);
+
+  // all values are selected as default
+  const [selectedNodeLabels, setSelectedNodeLabels] =
+    useState<(string | undefined)[]>(labels);
+  // show/hide node labels button
+  const [showNodeLabels, setShowNodeLabels] = useState<boolean>(false);
+
+  const ref = useRef<any>(null);
+
+  // custom switch based on MUI
+  const AntSwitch = withStyles((theme: Theme) =>
+    createStyles({
+      root: {
+        width: 28,
+        height: 16,
+        padding: 0,
+        display: 'flex',
+      },
+      switchBase: {
+        padding: 2,
+        color: theme.palette.grey[500],
+        '&$checked': {
+          transform: 'translateX(12px)',
+          color: theme.palette.common.white,
+          '& + $track': {
+            opacity: 1,
+            backgroundColor: theme.palette.primary.main,
+            borderColor: theme.palette.primary.main,
+          },
+        },
+      },
+      thumb: {
+        width: 12,
+        height: 12,
+        boxShadow: 'none',
+      },
+      track: {
+        border: `1px solid ${theme.palette.grey[500]}`,
+        borderRadius: 16 / 2,
+        opacity: 1,
+        backgroundColor: theme.palette.common.white,
+      },
+      checked: {},
+    })
+  )(Switch);
+
+  return (
+    <>
+      <NetworkPlot
+        containerStyles={{ width: DEFAULT_PLOT_SIZE }}
+        selectedNodeLabels={selectedNodeLabels}
```

Comment:
i think it may have been me who suggested "selectedNodeLabels", but i just realized we already use "selected" to mean something different (when a user has clicked on a node). So instead let's change this prop name to `visibleNodeLabels` or something like that.

Reply:
@asizemore changed to visibleNodeLabels as suggested